pkgs/tools/backup/bupstash/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchurl, rustc, cargo, ronn, pkg-config, libsodium }:
+
+stdenv.mkDerivation rec {
+  pname = "bupstash";
+  version = "0.6.3";
+
+  src = fetchurl {
+    url    = "https://github.com/andrewchambers/bupstash/releases/download/v${version}/bupstash-v${version}-src+deps.tar.gz";
+    sha256 = "0cvh50ar43wl0lbvck3bd315p6ffz7qspy31wk208p1anzk56x46";
+  };
+
+  buildInputs = [ rustc cargo ronn pkg-config libsodium ];
+
+  unpackPhase = ''
+    tar -xf "$src"
+  '';
+
+  buildPhase = ''
+    export RUBYOPT="-KU -E utf-8:utf-8"
+    ronn doc/man/*.md
+    cargo build --offline --release
+  '';
+
+  installPhase = ''
+    mkdir -p "$out"/{bin,share/man}
+    cp doc/man/*.1 "$out"/share/man
+    cp doc/man/*.7 "$out"/share/man
+    cp target/release/bupstash "$out"/bin
+  '';
+
+  doCheck = false;
```

Comment:
Why did you disable tests?

Reply:
There is an open issue that causes the full test suite to occasionally fail in a harmless way (timing problem), I don't want to propagate this failure to the package.

Reply:
I have left the cargo tests enabled as they pass without issue now. (The cli test suite is still not enable.)